bigquery/google/cloud/bigquery/job.py

```diff
@@ -1035,6 +1047,15 @@ def _populate_config_resource(self, configuration):
                 {udf_resource.udf_type: udf_resource.value}
                 for udf_resource in self._udf_resources
             ]
+        if len(self._query_parameters) > 0:
+            configuration[self._QUERY_PARAMETERS_KEY] = [
+                query_parameter.to_api_repr()
+                for query_parameter in self._query_parameters
+            ]
+            if self._query_parameters[0].name is None:
```

Comment:
The assumption is that every query param in `self._query_parameters` have the same mode. What enforces that assumption?